langchain_dappier/retrievers.py

```
@@ -0,0 +1,212 @@
+            dp_client = Dappier(api_key=self.api_key or os.environ["DAPPIER_API_KEY"])
+            response = dp_client.get_ai_recommendations(
+                query=query,
+                data_model_id=self.data_model_id,
+                similarity_top_k=kwargs.get("k", self.k),
+                ref=self.ref,
+                num_articles_ref=self.num_articles_ref,
+                search_algorithm=self.search_algorithm,
+            )
+            return self._extract_documents(response=response)
+        except Exception as e:
+            raise ValueError(f"Error while retrieving documents: {e}") from e
```

Ruff (0.8.2)

Avoid specifying long messages outside the exception class

(TRY003)